### DISCUSSION TITLE
[security]Introduce row/column level security

### DISCUSSION THREAD

**Discussion by GrapeBaBa** [databendlabs/databend#3546] [Ideas]
Security is an important capability for cloud data system. Row level security, Column level security and dynamic data mask are quite useful for data sharing and data governance use case.

I would like to introduce these features to databend. After I read the snowflake document https://docs.snowflake.com/en/user-guide/security-row-intro.html#how-does-a-row-access-policy-work, I think it can be implemented in the current codebase not difficult. 

Hope some thoughts and suggestions.

**Comment by bohutang** [comment]
Yes, we(@flaneur2020) are doing some fundamental work on it.
We define out some objects: database, table, share, role etc, and bind some privileges to these objects.
For example:
```
GRANT SELECT(x,y) ON TABLE db.table TO USER bohu
// SELECT(x,y) is privileges
// db.table is table object
// bohu is user object
```
RBAC is tracked here: https://github.com/datafuselabs/databend/issues/2815
There is still a lot of work to be done, but it is a very interesting feature!
